tests/conftest.py

```diff
@@ -23,6 +23,15 @@
     pass  # so the environment without spark doesn't break
 
 
+@pytest.fixture
```

Comment:
I also like the context manager option, maybe it can be implemented as we are discussing here: https://github.com/Microsoft/Recommenders/issues/701#issuecomment-479436972

Reply:
Let me try out that. looks promising.

Reply:
checked and turns out normal fixture does the same thing since newer version of pytest.
> Since pytest-3.0, fixtures using the normal fixture decorator can use a yield statement to provide fixture values and execute teardown code, exactly like yield_fixture in previous versions. Marking functions as yield_fixture is still supported, but deprecated and should not be used in new code.

Reply:
good, so if we use the fixture as you program it using: `with temp_dir: ` is the folder erased after the test function is finished?

Reply:
**yes**. Even try/finally does remove the dir too, but using the context manager made the code simpler and give us peace of mind :-)